packages/story-editor/src/components/header/buttons/buttonWithChecklistWarning.js

```diff
@@ -72,18 +75,23 @@ InnerButton.propTypes = {
 };
 
 function ButtonWithChecklistWarning({
-  text,
-  isUploading,
-  canPublish,
+  disabled,
+  hasFutureDate,
   ...buttonProps
 }) {
+  const isUploading = useIsUploadingToStory();
+  const { isSaving, canPublish, status } = useStory(({ state }) => ({
+    isSaving: state.meta?.isSaving,
```

Comment:
Some unit tests were having trouble with this, so checking if `isSaving` exists to keep tests intact